src/main/java/com/google/devtools/build/lib/pkgcache/LoadingOptions.java

```diff
@@ -56,32 +58,38 @@
 
   @Option(
     name = "build_tag_filters",
-    converter = CommaSeparatedOptionListConverter.class,
+    converter = ExpressionConverter.class,
     defaultValue = "",
     documentationCategory = OptionDocumentationCategory.UNCATEGORIZED,
     effectTags = {OptionEffectTag.UNKNOWN},
     help =
-        "Specifies a comma-separated list of tags. Each tag can be optionally "
-            + "preceded with '-' to specify excluded tags. Only those targets will be built that "
-            + "contain at least one included tag and do not contain any excluded tags. This option "
-            + "does not affect the set of tests executed with the 'test' command; those are be "
-            + "governed by the test filtering options, for example '--test_tag_filters'"
+        "Specifies a boolean expression or a list of tags separated by comma. A boolean expression must follow the "
+            + "syntax: expr = IDENT | expr 'or' expr | expr 'and' expr | 'not' expr | '(' expr ')'. "
+            + "When specifying a list of tags a tag can be optionally preceded with '-' to specify excluded tags. "
+            + "Only those targets will be built that contain at least one included tag and do not contain any "
+            + "excluded tags. This option does not affect the set of tests executed with the 'test' command; "
+            + "those are to be governed by the test filtering options, for example '--test_tag_filters'"
+            + "Example of boolean expression: tag1 and (tag2 or not tag3). "
+            + "Example of list of tags: tag1, tag2, -tag3. "
   )
-  public List<String> buildTagFilterList;
+  public Expression buildTagFilter;
 
   @Option(
     name = "test_tag_filters",
-    converter = CommaSeparatedOptionListConverter.class,
+    converter = ExpressionConverter.class,
     defaultValue = "",
     documentationCategory = OptionDocumentationCategory.UNCATEGORIZED,
     effectTags = {OptionEffectTag.UNKNOWN},
     help =
-        "Specifies a comma-separated list of test tags. Each tag can be optionally "
-            + "preceded with '-' to specify excluded tags. Only those test targets will be "
-            + "found that contain at least one included tag and do not contain any excluded "
-            + "tags. This option affects --build_tests_only behavior and the test command."
+        "Specifies a boolean expression or a list of tags separated by comma. A boolean expression must follow the "
```

Comment:
Don't duplicate the comment. Instead, direct the reader to --build_tag_filters. 

(Perhaps better the other way around: I suspect tags are used more for tests than builds.)